### PR TITLE
Add CRUD to MeterReading

### PR DESCRIPTION
**Pull request by nllong**
#### Any background context you want to provide?

Meter readings can only be imported from data files (green button, BuildingWync, or specifically formatted ESPM formats). Sometimes these files are not available for properties and users will want to import the data via a direct API call. 

#### What's this PR do?
* Adds in the MeterReading view set for creating individual meter readings on a meter
* Create 3 new meter types to handle generic meter types: electric, district chilled water, and custom meter. Prior to this change the district chilled water also included how the water was created, which can be unknown.

#### How should this be manually tested?
* The API methods are not available to the front end. Testing this is easiest through swagger, postman, and/or curl. The unit tests cover the functionality.
* See /v3/properties/{property_pk}/meters/{meter_pk}/readings/ in Swagger

#### What are the relevant tickets?
#3756

### CODE REVIEW

File: seed/api/v3/urls.py [hunk 1]
```diff
@@ -107,6 +108,9 @@
 properties_router.register(r'meters', MeterViewSet, basename='property-meters')
 properties_router.register(r'notes', NoteViewSet, basename='property-notes')
 properties_router.register(r'scenarios', PropertyScenarioViewSet, basename='property-scenarios')
+# This is a third level router, so we need to register it with the second level router
+meters_router = nested_routers.NestedSimpleRouter(properties_router, r'meters', lookup='meter')
```
Comment (nllong) [inline seed/api/v3/urls.py:112]: @anchapin  -- here is the example of the third level route which will also need to be created for /properties/1/scenarios/10/measures

File: seed/views/v3/meter_readings.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        # the permissions for the user
+        org_id = self.get_organization(self.request)
+        # get the property id - since the meter is associated with the property (not the property view)
+
+        # even though it is named 'property_pk' it is really the property view id
+        property_view_pk = self.kwargs.get('property_pk', None)
+        if not property_view_pk:
+            # Return None otherwise swagger will not be able to process the request
+            return MeterReading.objects.none()
+        property_view = PropertyView.objects.get(pk=property_view_pk)
```
Comment (haneslinger) [inline seed/views/v3/meter_readings.py:33]: super nit: this should probably be try/catch-ed

File: seed/views/v3/meter_readings.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        # Grab the meter id
+        meter_pk = self.kwargs.get('meter_pk', None)
+        if not meter_pk:
+            # Return None otherwise swagger will not be able to process the request
+            return MeterReading.objects.none()
+        self.meter_pk = meter_pk
+
+        return MeterReading.objects.filter(
+            meter__property__organization_id=org_id, meter__property=self.property_pk, meter_id=meter_pk
+        )
```
Comment (haneslinger) [inline seed/views/v3/meter_readings.py:45]: again, try/catch-ed

File: seed/tests/test_meter_views.py [hunk 4]
```diff
@@ -147,3 +145,102 @@ def test_update_meter(self):
...
+        }
+
+        response = self.client.post(url, data=json.dumps(payload), content_type='application/json')
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response.json()['reading'], 10)
+
+        # now delete the item and verify that there are no more readings in the database
+        detail_url = reverse('api:v3:property-meter-readings-detail', kwargs={'property_pk': property_view.id, 'meter_pk': meter_pk, 'pk': '2022-01-05T05:00:00Z'})
+        response = self.client.get(detail_url, content_type='application/json')
+        self.assertEqual(response.status_code, 200)
```
Comment (haneslinger) [inline seed/tests/test_meter_views.py:238]: what are these lines doing for the test?

**Review by haneslinger — COMMENTED** [review]
Looking good. What does the Update endpoint look like? Can you update just one reading or must you do them are?

File: seed/views/v3/meter_readings.py (hunk 3 above)
Comment (nllong) [reply seed/views/v3/meter_readings.py:45]: This is a filter, so I don't know the advantage of try catching this unless you are concerned about some sort of syntax error. We are verifying above the call that property_pk and meter_pk exist, so the result would just be an empty MeterReading set... which is okay.

File: seed/tests/test_meter_views.py (hunk 4 above)
Comment (nllong) [reply seed/tests/test_meter_views.py:279]: Verifies that the data are returned, but with just a 200. Nothing fancy.

File: seed/views/v3/meter_readings.py (hunk 2 above)
Comment (nllong) [reply seed/views/v3/meter_readings.py:33]: updated...

**Comment by nllong** [comment]
> Looking good. What does the Update endpoint look like? Can you update just one reading or must you do them are?

Yup, i added both individual updates and bulk updates. The Swagger page is probably the best view on what one can do now with the meter readings. The one oddity is that since the data are part of the timescale hypertable, the start_time is the primary key.

image

File: seed/lib/superperms/orgs/models.py [hunk 5]
```diff
@@ -161,9 +143,9 @@ class Organization(models.Model):
         'Fuel Oil (No. 5 and No. 6)': 'kBtu (thousand Btu)',
         'Kerosene': 'kBtu (thousand Btu)',
         'Natural Gas': 'kBtu (thousand Btu)',
-        'Other:': 'kBtu (thousand Btu)',
+        'Other:': 'kBtu (thousand Btu)',  # yes, other has a colon at the end.
```
Comment (axelstudios) [inline seed/lib/superperms/orgs/models.py:146]: 🥴

**Review by axelstudios — CHANGES_REQUESTED** [review]
500 errors can result from meter endpoints

**Review by axelstudios — APPROVED** [review]

**Pull request merged**